1.15 fails to compile on Linux aarch64

On openSUSE Tumbleweed (packaged at https://build.opensuse.org/package/show/devel:languages:haxe/hashlink ) we are getting this compilation error on aarch64
```
[   50s] [ 62%] Linking C shared library bin/libhl.so
[   50s] /usr/bin/cmake -E cmake_link_script CMakeFiles/libhl.dir/link.txt --verbose=1
[   50s] /tmp/ccV8o7PV.s: Assembler messages:
[   50s] /tmp/ccV8o7PV.s:31: Error: unknown mnemonic `int3' -- `int3'
[   50s] /tmp/ccV8o7PV.s:83: Error: unknown mnemonic `int3' -- `int3'
[   50s] /tmp/ccV8o7PV.s:12537: Error: unknown mnemonic `int3' -- `int3'
[   50s] /tmp/ccV8o7PV.s:12954: Error: unknown mnemonic `int3' -- `int3'
[   50s] make[3]: *** [/tmp/ccIDgu5H.mk:11: /tmp/ccHH8To9.ltrans3.ltrans.o] Error 1
[   50s] make[3]: *** Waiting for unfinished jobs....
[   50s] lto-wrapper: fatal error: make returned 2 exit status
[   50s] compilation terminated.
[   50s] /usr/lib64/gcc/aarch64-suse-linux/14/../../../../aarch64-suse-linux/bin/ld: error: lto-wrapper failed
[   50s] collect2: error: ld returned 1 exit status
```

The issue is with the definition of hl_debug_break() at https://github.com/HaxeFoundation/hashlink/blob/1.15/src/hl.h#L286-L301 . Previously it was only enabled if __i386__ was defined, but it was enabled for all architectures in #695

--- a/src/hl.h
+++ b/src/hl.h
@@ -283,28 +283,40 @@ C_FUNCTION_END
 C_FUNCTION_BEGIN
 HL_API void hl_debug_break( void );
 C_FUNCTION_END
-#elif defined(HL_LINUX)
-#	ifdef HL_64
-#	define hl_debug_break() \
-		if( hl_detect_debugger() ) \
-			__asm__("0: int3;" \
-			    ".pushsection embed-breakpoints;" \
-			    ".quad 0b;" \
-			    ".popsection")
-#	else
-#	define hl_debug_break() \
-		if( hl_detect_debugger() ) \
-			__asm__("0: int3;" \
-			    ".pushsection embed-breakpoints;" \
-			    ".long 0b;" \
-			    ".popsection")
+#elif !defined(HL_CONSOLE)
+
+// use __builtin_debugtrap when available
+// fall back to breakpoint instructions for certain architectures
+// else raise SIGTRAP
+#	ifdef __has_builtin
+#	if __has_builtin(__builtin_debugtrap)
+#	define USE_BUILTIN_DEBUG_TRAP 1
 #	endif
-#elif defined(HL_MAC)
-#include <signal.h>
-#include <mach/mach.h>
+#	endif
+
+#	ifdef USE_BUILTIN_DEBUG_TRAP
 #	define hl_debug_break() \
 		if( hl_detect_debugger() ) \
-			raise(SIGTRAP);//__builtin_trap();
+			__builtin_debugtrap()
+#	elif defined(__x86_64__) || defined(__i386__)
+#	define hl_debug_break() \
+		if( hl_detect_debugger() ) \
+			__asm__("int3;")
+#	elif defined(__aarch64__)
+#	define hl_debug_break() \
+		if( hl_detect_debugger() ) \
+			__asm__("brk #0xf000;")
+#	elif defined(__riscv)
+#	define hl_debug_break() \
+		if( hl_detect_debugger() ) \
+			__asm__("ebreak;")
+#	else
+#	include <signal.h>
+#	define hl_debug_break() \
+		if( hl_detect_debugger() ) \
+			raise(SIGTRAP)
+#	endif
+#undef USE_BUILTIN_DEBUG_TRAP
 #else
 #	define hl_debug_break()
 #endif
@@ -912,6 +924,11 @@ struct _hl_trap_ctx {
 #define HL_TRACK_MASK		(HL_TRACK_ALLOC | HL_TRACK_CAST | HL_TRACK_DYNFIELD | HL_TRACK_DYNCALL)
 
 #define HL_MAX_EXTRA_STACK 64
+
+#ifdef HL_MAC
+#include <mach/mach.h>
+#include <signal.h>
+#endif
 
 typedef struct {
 	int thread_id;

